runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java

```diff
@@ -1024,11 +1026,39 @@ else if (limit > endOfMethodLimit)
 
                 if (isCorsPreflightRequest(headers))
                 {
+                    HttpEventFW event = server.httpEventRW
+                        .wrap(server.eventBuffer, 0, server.eventBuffer.capacity())
+                        .corsPreflightRequestNotAllowed(e -> e
+                            .originId(server.originId)
+                            .routedId(server.routedId)
+                            .initialId(server.initialId)
+                            .replyId(server.replyId)
+                            .affinity(server.affinity)
+                            .authorization(authorization)
+                            .budgetId(budgetId)
+                            .traceId(traceId)
+                        )
+                        .build();
+                    System.out.println(event); // TODO: Ati
                     server.onDecodeCorsPreflight(traceId, authorization, headers);
                     server.decoder = decodeEmptyLines;
                 }
                 else if (!isCorsRequestAllowed(server.binding, headers))
                 {
+                    HttpEventFW event = server.httpEventRW
+                        .wrap(server.eventBuffer, 0, server.eventBuffer.capacity())
+                        .corsRequestNotAllowed(e -> e
+                            .originId(server.originId)
+                            .routedId(server.routedId)
+                            .initialId(server.initialId)
+                            .replyId(server.replyId)
+                            .affinity(server.affinity)
+                            .authorization(authorization)
+                            .budgetId(budgetId)
+                            .traceId(traceId)
+                        )
+                        .build();
+                    System.out.println(event); // TODO: Ati
```

Comment:
We'll need an API on `EngineContext` to log the event.

Reply:
pls chk